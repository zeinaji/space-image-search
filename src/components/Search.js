import React, { useState } from "react";
import "../styles/Search.css";
import getImages from "../requests/getImages";

const Search = ({ setSearchResults }) => {
  const [value, setValue] = useState("");
  const handleSubmit = (event) => {
    event.preventDefault();
    const images = getImages(value);
    setSearchResults(images);
  };

  return (
    <div className="Search">
      <form className="form-wrapper" onSubmit={handleSubmit}>
        <input
          className="search-input"
          type="text"
          onChange={(e) => setValue(e.target.value)}
        />
        <button className="submit" type="submit">
          Go!
        </button>
      </form>
    </div>
  );
};

export default Search;
