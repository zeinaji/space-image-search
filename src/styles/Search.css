.nasa-logo {
  height: 300px;
  width: 300px;
  margin-bottom: -20px;
}

.search-input,
.submit {
  border: 1px solid grey;
  vertical-align: middle;
}

.submit {
  margin-top: 0;
  width: 45px;
  height: 23px;
  background-color: rgb(199, 46, 46);
  color: white;
  font-family: "Comic Sans MS", cursive, sans-serif;
  border-radius: 0px 7px 7px 0px;
  border-left: none;
  cursor: pointer;
}

.search-input {
  height: 19px;
  border-radius: 7px 0px 0px 7px;
  border-right: none;
}

button:focus,
input:focus {
  outline: none;
}
