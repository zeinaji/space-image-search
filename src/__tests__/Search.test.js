import React from "react";
import { render, screen } from "@testing-library/react";
import Search from "../components/Search";

describe("Search", () => {
  xit("it renders correctly", () => {
    const { asFragment } = render(<Search />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders input box", () => {
    render(<Search />);
    document.querySelector(".search-input").value = "mockValue";
    const inputValue = screen.getByDisplayValue("mockValue");

    expect(inputValue).toBeTruthy();
  });

  it("renders button", () => {
    render(<Search />);
    expect(screen.getByRole("button")).toHaveClass("submit");
  });
});
