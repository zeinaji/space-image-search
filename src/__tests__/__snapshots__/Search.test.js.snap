// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search it renders correctly 1`] = `
<DocumentFragment>
  <div
    class="Search"
  >
    <form
      class="form-wrapper"
    >
      <input
        class="search-input"
        type="text"
      />
      <button
        class="submit"
        type="submit"
      >
        Go!
      </button>
    </form>
  </div>
</DocumentFragment>
`;
